statistics/update.go

```diff
@@ -133,3 +137,62 @@ func (h *Handle) dumpTableStatDeltaToKV(id int64, delta variable.TableDelta) err
 	_, err = h.ctx.(sqlexec.SQLExecutor).Execute("commit")
 	return errors.Trace(err)
 }
+
+const (
+	// StatsOwnerKey is the stats owner path that is saved to etcd.
+	StatsOwnerKey = "/tidb/stats/owner"
+	// StatsPrompt is the prompt for stats owner manager.
+	StatsPrompt = "stats"
+)
+
+func needAnalyzeTable(tbl *Table, limit time.Duration) bool {
+	if tbl.ModifyCount == 0 {
+		return false
+	}
+	t := time.Unix(0, oracle.ExtractPhysical(tbl.Version)*int64(time.Millisecond))
+	if time.Now().Sub(t) < limit {
+		return false
+	}
+	for _, col := range tbl.Columns {
+		if len(col.Buckets) > 0 {
+			return false
+		}
+	}
+	for _, idx := range tbl.Indices {
+		if len(idx.Buckets) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// HandleAutoAnalyze analyzes the newly created table or index.
+func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) error {
+	dbs := is.AllSchemaNames()
+	for _, db := range dbs {
+		tbls := is.SchemaTables(model.NewCIStr(db))
+		for _, tbl := range tbls {
+			tblInfo := tbl.Meta()
+			statsTbl := h.GetTableStats(tblInfo.ID)
+			if statsTbl.Pseudo || statsTbl.Count == 0 {
```

Comment:
1. First, There is no direct mapping from table_id to table_name, which would make it hard to produce the analyze statement. Second, modify_count >= count does not ensure that this is a newly created table. Third, Since I will implement a job queue, the job queue could be a priority queue, which I think is a better solution.
2. Well, I did not notice it before. A deleted table has non-zero count is reasonable. So, let's directly make the deleted table's count 0.